src/payment/strategies/digitalriver/digitalriver-payment-strategy.spec.ts

```diff
@@ -0,0 +1,259 @@
+                    message: 'message',
+                }],
+            });
+            expect(options.digitalriver?.onError).toBeCalled();
+            expect(digitalRiverLoadResponse.createDropin).toBeCalled();
+        });
+
+        it('throws an error when load response is empty or not provided', () => {
+            const promise = strategy.initialize(options);
+            jest.spyOn(digitalRiverScriptLoader, 'load').mockReturnValue(Promise.resolve(undefined));
+
+            return expect(promise).rejects.toThrow(NotInitializedError);
+        });
+
+        it('throws an error when DigitalRiver options is not provided', () => {
+            const error = new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+            options.digitalriver = undefined;
+            const promise = strategy.initialize(options);
+
+            return expect(promise).rejects.toThrow(error);
+        });
+
+        it('throws an error when DigitalRiver clientToken is not provided', () => {
+            const error = new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            paymentMethodMock = {...getPaymentMethod(), clientToken: ''};
+            loadPaymentMethodAction = of(createAction(PaymentMethodActionType.LoadPaymentMethodSucceeded, paymentMethodMock, {methodId: paymentMethodMock.id}));
+            const promise = strategy.initialize(options);
+
+            return expect(promise).rejects.toThrow(error);
+        });
+
+        it('throws an error when DigitalRiver clientToken is not receiving correct data ', () => {
+            const error = new Error('Unexpected token o in JSON at position 0');
+            paymentMethodMock = {...getPaymentMethod(), clientToken: 'ok'};
+            loadPaymentMethodAction = of(createAction(PaymentMethodActionType.LoadPaymentMethodSucceeded, paymentMethodMock, {methodId: paymentMethodMock.id}));
+            const promise = strategy.initialize(options);
+
+            return expect(promise).rejects.toThrow(error);
+        });
+
+        it('throws an error when customer is not provided', () => {
+            const promise = strategy.initialize(options);
+            const error = new NotInitializedError(NotInitializedErrorType.CustomerNotInitialized);
+            jest.spyOn(store.getState().customer, 'getCustomer').mockReturnValue(undefined);
+
+            return expect(promise).rejects.toThrow(error);
+        });
+
+        it('throws an error when data on onSuccess event is not provided', async () => {
+            jest.spyOn(digitalRiverLoadResponse, 'createDropin').mockImplementation(({onSuccess}) => {
+                try {
+                    onSuccess(undefined);
+                } catch (error) {
+                    expect(true).toBeTruthy();
+                }
+
+                return digitalRiverComponent;
+            });
+            await strategy.initialize(options);
+            expect(true).toBeTruthy();
+        });
+    });
```

Comment:
This test seems to be wrong since you're expecting true to be true. so please correct this test with a most appropriate expect